pkg/bitcoin_difficulty/chain.go

```diff
@@ -0,0 +1,34 @@
+package bitcoin_difficulty
+
+import (
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
+)
+
+// Chain defines the interface that pertains specifically to the process of
+// updating Bitcoin difficulty.
+type Chain interface {
+	// Ready checks whether the relay is active (i.e. genesis has been
+	// performed).
```

Comment:
Please keep in mind [this limitation](https://github.com/keep-network/tbtc-v2/blob/83310bdc9ed934e286bc9ea5091cc16979950134/solidity/contracts/relay/LightRelay.sol#L130-L132): 

```
    /// @dev If the relay is used by querying the current and previous epoch
    /// difficulty, at least one retarget needs to be provided after genesis;
    /// otherwise the prevEpochDifficulty will be uninitialised and zero.
```

We should make it clear in the docs for `Ready`. `Ready` returning `true` does not mean `getPrevEpochDifficulty` will give a meaningful result.

Reply:
Done in 37426409216d716c85a610028748547a963172d7.